diff -Naur qt-everywhere-opensource-src-5.3.2/qtbase/mkspecs/features/mac/objective_c.prf qt-src-5.3.2/qtbase/mkspecs/features/mac/objective_c.prf
--- qt-everywhere-opensource-src-5.3.2/qtbase/mkspecs/features/mac/objective_c.prf	2014-09-11 20:48:07.000000000 +1000
+++ qt-src-5.3.2/qtbase/mkspecs/features/mac/objective_c.prf	2016-03-17 14:34:52.000000000 +1100
@@ -19,5 +19,6 @@
 objective_c.output = $$OBJECTIVE_C_OBJECTS_DIR/${QMAKE_FILE_BASE}$${first(QMAKE_EXT_OBJ)}
 objective_c.input = OBJECTIVE_SOURCES
 objective_c.name = Compile ${QMAKE_FILE_IN}
+objective_c.skip_preprocess = 1
 silent:objective_c.commands = @echo objective-c ${QMAKE_FILE_IN} && $$objective_c.commands
 QMAKE_EXTRA_COMPILERS += objective_c
diff -Naur qt-everywhere-opensource-src-5.3.2/qtbase/qmake/generators/mac/pbuilder_pbx.cpp qt-src-5.3.2/qtbase/qmake/generators/mac/pbuilder_pbx.cpp
--- qt-everywhere-opensource-src-5.3.2/qtbase/qmake/generators/mac/pbuilder_pbx.cpp	2014-09-11 20:48:00.000000000 +1000
+++ qt-src-5.3.2/qtbase/qmake/generators/mac/pbuilder_pbx.cpp	2016-03-17 14:34:52.000000000 +1100
@@ -779,6 +779,8 @@
                 mkt << "compilers:";
                 const ProStringList &compilers = project->values("QMAKE_EXTRA_COMPILERS");
                 for(int compiler = 0; compiler < compilers.size(); ++compiler) {
+                   if(!project->isEmpty(ProKey(compilers.at(compiler).toQString() + ".skip_preprocess")))
+                       continue;
                     const ProStringList &tmp_out = project->values(ProKey(compilers.at(compiler) + ".output"));
                     if (tmp_out.isEmpty())
                         continue;
diff -Naur qt-everywhere-opensource-src-5.3.2/qtbase/src/plugins/platforms/cocoa/qcocoacursor.mm qt-src-5.3.2/qtbase/src/plugins/platforms/cocoa/qcocoacursor.mm
--- qt-everywhere-opensource-src-5.3.2/qtbase/src/plugins/platforms/cocoa/qcocoacursor.mm	2014-09-11 20:48:06.000000000 +1000
+++ qt-src-5.3.2/qtbase/src/plugins/platforms/cocoa/qcocoacursor.mm	2016-03-17 16:21:47.000000000 +1100
@@ -81,7 +81,7 @@
     pos.x = position.x();
     pos.y = position.y();

-    CGEventRef e = CGEventCreateMouseEvent(0, kCGEventMouseMoved, pos, 0);
+    CGEventRef e = CGEventCreateMouseEvent(0, kCGEventMouseMoved, pos, kCGMouseButtonLeft);
     CGEventPost(kCGHIDEventTap, e);
     CFRelease(e);
 }
diff -Naur qt-everywhere-opensource-src-5.3.2/qtbase/src/tools/qlalr/lalr.cpp qt-src-5.3.2/qtbase/src/tools/qlalr/lalr.cpp
--- qt-everywhere-opensource-src-5.3.2/qtbase/src/tools/qlalr/lalr.cpp	2014-09-11 20:48:01.000000000 +1000
+++ qt-src-5.3.2/qtbase/src/tools/qlalr/lalr.cpp	2016-03-17 16:14:01.000000000 +1100
@@ -246,11 +246,11 @@
   non_terminals.insert (accept_symbol);
 }

-struct _Nullable: public std::unary_function<Name, bool>
+struct Nullable: public std::unary_function<Name, bool>
 {
   Automaton *_M_automaton;

-  _Nullable (Automaton *aut):
+  Nullable (Automaton *aut):
     _M_automaton (aut) {}

   bool operator () (Name name) const
@@ -308,7 +308,7 @@

       for (RulePointer rule = _M_grammar->rules.begin (); rule != _M_grammar->rules.end (); ++rule)
         {
-          NameList::iterator nn = std::find_if (rule->rhs.begin (), rule->rhs.end (), std::not1 (_Nullable (this)));
+          NameList::iterator nn = std::find_if (rule->rhs.begin (), rule->rhs.end (), std::not1 (Nullable (this)));

           if (nn == rule->rhs.end ())
             changed |= nullables.insert (rule->lhs).second;
@@ -643,7 +643,7 @@
                   if (! _M_grammar->isNonTerminal (*A))
                     continue;

-                  NameList::iterator first_not_nullable = std::find_if (dot, rule->rhs.end (), std::not1 (_Nullable (this)));
+                  NameList::iterator first_not_nullable = std::find_if (dot, rule->rhs.end (), std::not1 (Nullable (this)));
                   if (first_not_nullable != rule->rhs.end ())
                     continue;

diff -Naur qt-everywhere-opensource-src-5.3.2/qtmultimedia/src/plugins/avfoundation/mediaplayer/avfmediaplayersession.mm qt-src-5.3.2/qtmultimedia/src/plugins/avfoundation/mediaplayer/avfmediaplayersession.mm
--- qt-everywhere-opensource-src-5.3.2/qtmultimedia/src/plugins/avfoundation/mediaplayer/avfmediaplayersession.mm	2014-09-11 20:48:26.000000000 +1000
+++ qt-src-5.3.2/qtmultimedia/src/plugins/avfoundation/mediaplayer/avfmediaplayersession.mm	2016-03-18 09:32:21.000000000 +1100
@@ -295,7 +295,7 @@
     //AVPlayerItem "status" property value observer.
     if (context == AVFMediaPlayerSessionObserverStatusObservationContext)
     {
-        AVPlayerStatus status = [[change objectForKey:NSKeyValueChangeNewKey] integerValue];
+        AVPlayerStatus status = (AVPlayerStatus)[[change objectForKey:NSKeyValueChangeNewKey] integerValue];
         switch (status)
         {
             //Indicates that the status of the player is not yet known because
