diff -Naur qt-everywhere-opensource-src-5.6.0/qtwebengine/src/3rdparty/chromium/third_party/skia/src/core/SkEdge.cpp qt-src-5.6.0/qtwebengine/src/3rdparty/chromium/third_party/skia/src/core/SkEdge.cpp
--- qt-everywhere-opensource-src-5.6.0/qtwebengine/src/3rdparty/chromium/third_party/skia/src/core/SkEdge.cpp	2016-03-03 17:48:36.000000000 +1100
+++ qt-src-5.6.0/qtwebengine/src/3rdparty/chromium/third_party/skia/src/core/SkEdge.cpp	2016-04-04 12:32:50.098280200 +1000
@@ -228,7 +228,7 @@
 
     fWinding    = SkToS8(winding);
     //fCubicDShift only set for cubics
-    fCurveCount = SkToS8(1 << shift);
+	fCurveCount = SkToS8(1i64 << shift);
 
     /*
      *  We want to reformulate into polynomial form, to make it clear how we
