diff -Naur qt-everywhere-opensource-src-5.5.0/qtbase/mkspecs/features/mac/objective_c.prf qt-5.5.0-src/qtbase/mkspecs/features/mac/objective_c.prf
--- qt-everywhere-opensource-src-5.5.0/qtbase/mkspecs/features/mac/objective_c.prf	2015-06-30 06:05:10.000000000 +1000
+++ qt-5.5.0-src/qtbase/mkspecs/features/mac/objective_c.prf	2015-07-03 10:41:10.000000000 +1000
@@ -19,5 +19,6 @@
 objective_c.output = $$OBJECTIVE_C_OBJECTS_DIR/${QMAKE_FILE_BASE}$${first(QMAKE_EXT_OBJ)}
 objective_c.input = OBJECTIVE_SOURCES
 objective_c.name = Compile ${QMAKE_FILE_IN}
+objective_c.skip_preprocess = 1
 silent:objective_c.commands = @echo objective-c ${QMAKE_FILE_IN} && $$objective_c.commands
 QMAKE_EXTRA_COMPILERS += objective_c
diff -Naur qt-everywhere-opensource-src-5.5.0/qtbase/qmake/generators/mac/pbuilder_pbx.cpp qt-5.5.0-src/qtbase/qmake/generators/mac/pbuilder_pbx.cpp
--- qt-everywhere-opensource-src-5.5.0/qtbase/qmake/generators/mac/pbuilder_pbx.cpp	2015-06-30 06:03:23.000000000 +1000
+++ qt-5.5.0-src/qtbase/qmake/generators/mac/pbuilder_pbx.cpp	2015-07-03 10:42:00.000000000 +1000
@@ -749,6 +749,8 @@
                 mkt << "compilers:";
                 const ProStringList &compilers = project->values("QMAKE_EXTRA_COMPILERS");
                 for(int compiler = 0; compiler < compilers.size(); ++compiler) {
+                   if(!project->isEmpty(ProKey(compilers.at(compiler).toQString() + ".skip_preprocess")))
+                       continue;
                     const ProStringList &tmp_out = project->values(ProKey(compilers.at(compiler) + ".output"));
                     if (tmp_out.isEmpty())
                         continue;
diff -Naur qt-everywhere-opensource-src-5.5.0/qtbase/src/corelib/global/qcompilerdetection.h qt-5.5.0-src/qtbase/src/corelib/global/qcompilerdetection.h
--- qt-everywhere-opensource-src-5.5.0/qtbase/src/corelib/global/qcompilerdetection.h	2015-06-30 06:04:58.000000000 +1000
+++ qt-5.5.0-src/qtbase/src/corelib/global/qcompilerdetection.h	2015-07-03 10:43:10.000000000 +1000
@@ -155,7 +155,7 @@
 /* Clang also masquerades as GCC */
 #    if defined(__apple_build_version__)
 #      /* http://en.wikipedia.org/wiki/Xcode#Toolchain_Versions */
-#      if __apple_build_version__ >= 6020049
+#      if __apple_build_version__ >= 7000053
 #        define Q_CC_CLANG 306
 #      elif __apple_build_version__ >= 6000051
 #        define Q_CC_CLANG 305
